core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/TelemetryBuffer.java

```diff
@@ -55,25 +57,32 @@
      */
     private final class TelemetryBufferTelemetriesFetcher implements TelemetriesTransmitter.TelemetriesFetcher {
 
-        private final long expectedGeneration;
+        private final AtomicLong expectedGeneration;
 
-        private TelemetryBufferTelemetriesFetcher(long expectedGeneration) {
+        private TelemetryBufferTelemetriesFetcher(AtomicLong expectedGeneration) {
             this.expectedGeneration = expectedGeneration;
         }
 
+        // No need for low-level synchronization. Atomic long cover cover the generation check consistency
+        // ConcurrentLinkedList handles other parts for collection operation.
         @Override
         public Collection<T> fetch() {
-            synchronized (lock) {
```

Comment:
If we don't lock on the collection anymore, what is it guarding? Should it be removed?